alacritty_terminal/src/term/mod.rs

```diff
@@ -1572,15 +1572,15 @@ impl<T: EventListener> Handler for Term<T> {
         let num_cells = (self.cols() - destination).0;
 
         let line = cursor.point.line;
-        let line = &mut self.grid[line];
+        let line = &mut self.grid[line][..];
 
         for offset in (0..num_cells).rev() {
-            line[destination + offset] = line[source + offset].clone();
+            line.swap(destination.0 + offset, source.0 + offset);
         }
 
         // Cells were just moved out toward the end of the line;
         // fill in between source and dest with blanks.
-        for cell in &mut line[source..destination] {
+        for cell in &mut line[source.0..destination.0] {
             *cell = bg.into();
         }
```

Comment:
Can't these two be combined? It seems like a shame to iterate twice just to insert the blanks when really we already swap them into place.

Reply:
I've tried but it looked ugly a bit, since you should check for both boundaries in a loop, while here you check for them in a loop condition...

Reply:
Fair enough, it's worked well so far so that should be fine I suppose. Though I'd imagine having a check inside the loop would save us the increment/decrement of the counter we need to use with two loops, but compared to how expensive these loops are with the `swap` that is completely insignificant compared to possible code style advantages.